plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -156,56 +156,79 @@ private function get_location_context() {
 	}
 
 	/**
-	 * Enhances the Product Collection block with client-side pagination.
-	 *
-	 * This function identifies Product Collection blocks and adds necessary data attributes
-	 * to enable client-side navigation and animation effects. It also enqueues the Interactivity API runtime.
+	 * Attach the init directive to Product Collection block.
 	 *
 	 * @param string $block_content The HTML content of the block.
-	 * @param array  $block         Block details, including its attributes.
+	 * @param string $collection Collection type.
 	 *
-	 * @return string Updated block content with added interactivity attributes.
+	 * @return string Updated HTML content.
 	 */
-	public function enhance_product_collection_with_interactivity( $block_content, $block ) {
-		$is_product_collection_block    = $block['attrs']['query']['isProductCollectionBlock'] ?? false;
-		$is_enhanced_pagination_enabled = ! ( $block['attrs']['forcePageReload'] ?? false );
-		if ( $is_product_collection_block && $is_enhanced_pagination_enabled ) {
-			// Enqueue the Interactivity API runtime.
-			wp_enqueue_script( 'wc-interactivity' );
+	private function add_init_directive( $block_content, $collection ) {
+		$p = new \WP_HTML_Tag_Processor( $block_content );
 
-			$p = new \WP_HTML_Tag_Processor( $block_content );
+		// Add `data-init to the product collection block so we trigger JS event on render.
+		if ( $p->next_tag( array( 'class_name' => 'wp-block-woocommerce-product-collection' ) ) ) {
+			$p->set_attribute(
+				'data-wc-init',
+				'callbacks.onRender'
+			);
+			$p->set_attribute(
+				'data-wc-context',
+				wp_json_encode(
+					array(
+						'collection' => $collection,
+					),
+					JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_QUOT | JSON_HEX_AMP
+				)
+			);
```

Comment:
Super Nit: I am unsure if `data-wc-context` belongs to `add_init_directive` function. Probably, renaming this function will help make the code more readable 🙂

Reply:
I changed the function name to be more specific, good point Manish!